opengrok-indexer/src/main/java/org/opengrok/indexer/Metrics.java

```diff
@@ -22,19 +22,105 @@
+import org.opengrok.indexer.logger.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulates logic of meter registry setup and handling.
+ * Generally, the web application publishes metrics to Prometheus and the Indexer to StatsD.
+ */
 public final class Metrics {
 
-    private static final PrometheusMeterRegistry registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+    private static final Logger LOGGER = LoggerFactory.getLogger(Metrics.class);
+
+    private static final StatsdConfig statsdConfig = new StatsdConfig() {
+        @Override
+        public String get(String k) {
+            return null;
+        }
+
+        @Override
+        public StatsdFlavor flavor() {
+            String flavor = RuntimeEnvironment.getInstance().getStatsdConfig().getFlavor().toLowerCase();
+            switch (flavor) {
+                case "etsy":
+                    return StatsdFlavor.ETSY;
+                case "datadog":
+                    return StatsdFlavor.DATADOG;
+                case "sysdig":
+                    return StatsdFlavor.SYSDIG;
+                case "telegraf":
+                    return StatsdFlavor.TELEGRAF;
+            }
+
+            return null;
+        }
+
+        @Override
+        public int port() {
+            return RuntimeEnvironment.getInstance().getStatsdConfig().getPort();
+        }
+
+        @Override
+        public String host() {
+            return RuntimeEnvironment.getInstance().getStatsdConfig().getHost();
+        }
+
+        @Override
+        public boolean buffered() {
+            return true;
+        }
+    };
+
+    private static PrometheusMeterRegistry prometheusRegistry;
+    private static StatsdMeterRegistry statsdRegistry;
 
     static {
+        MeterRegistry registry;
+
+        if (RuntimeEnvironment.getInstance().getStatsdConfig().isEnabled()) {
+            LOGGER.log(Level.INFO, "configuring StatsdRegistry");
+
+            statsdRegistry = new StatsdMeterRegistry(statsdConfig, Clock.SYSTEM);
+
+            // Add tag for per-project reindex.
+            ArrayList<String> subFiles = RuntimeEnvironment.getInstance().getSubFiles();
+            if (subFiles.size() > 0) {
+                List<String> sList = subFiles.stream().
+                        map(s -> s.startsWith(Indexer.PATH_SEPARATOR_STRING) ? s.substring(1) : s).
+                        collect(Collectors.toList());
+                statsdRegistry.config().commonTags(Collections.singleton(Tag.of("projects",
+                        String.join(",", sList))));
+            }
+
+            registry = statsdRegistry;
+        } else {
+            LOGGER.log(Level.INFO, "configuring PrometheusRegistry");
+            prometheusRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
```

Comment:
Let's say that statsd is disabled. What is the point of creating Prometheus registry if it cannot be accessed?

Reply:
right, fixed.